test/testutils/client.go

```diff
@@ -0,0 +1,40 @@
+package testutils
+
+import (
+	"context"
+	"testing"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DeleteObject deletes obj from c.
+// Succeeds on success or if obj already did not exist in c.
+// Fails the test if calling Delete() returns an error other than NotFound.
+//
+// Does not ensure the object is actually deleted, just sends a delete request via the passed Client.
+// See #1596 for why this is non-trivial to guarantee.
```

Comment:
Do we want to just never delete objects if we cannot rely on successful deletes? Where are we relying on the delete behavior for proper functioning?